### PR TITLE
[TwigBridge][TwigBundle] Drop support for Twig 2

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 7.0
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

Let's make our lives easier and only support one major version of Twig in the future. Upgrading projects from Twig 2 to 3 shouldn't be much trouble anyway.

### CODE REVIEW

**Review by fabpot — APPROVED** [review]

**Review by GromNaN — APPROVED** [review]

**Review by stof — COMMENTED** [review]
The changelog file must be updated

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
         "symfony/uid": "^6.4|^7.0",
         "symfony/web-link": "^6.4|^7.0",
         "phpdocumentor/reflection-docblock": "^3.0|^4.0|^5.0",
-        "twig/twig": "^2.10|^3.0"
+        "twig/twig": "^3.0.4"
```
Comment (GromNaN) [inline src/Symfony/Bundle/FrameworkBundle/composer.json:73]: I think we should add a conflict rule, as is already done for symfony components in require-dev.

Comment (derrabus) [reply src/Symfony/Bundle/FrameworkBundle/composer.json:73]: Effectively, the http-kernel package does that for us currently. Shall I duplicate the rule in the bundles' composer.json file?

**Comment by derrabus** [comment]
> The changelog file must be updated

Done for TwigBundle and TwigBridge. Not sure if I need to do that for the other bundles/components as well.

**Comment by stof** [comment]
@derrabus I think it is OK as is.

File: src/Symfony/Component/VarDumper/composer.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
         "symfony/http-kernel": "^6.4|^7.0",
         "symfony/process": "^6.4|^7.0",
         "symfony/uid": "^6.4|^7.0",
-        "twig/twig": "^2.13|^3.0.4"
+        "twig/twig": "^3.0.4"
```
Comment (stof) [inline src/Symfony/Component/VarDumper/composer.json:28]: should this add the conflict rule ?

Comment (derrabus) [reply src/Symfony/Component/VarDumper/composer.json:28]: Not sure. It didn't have one before and I don't know why that is.

File: src/Symfony/Bundle/FrameworkBundle/composer.json (hunk 1 above)
Comment (GromNaN) [reply src/Symfony/Bundle/FrameworkBundle/composer.json:73]: You're right, the constraint is already imposed by http-kernel, which is required by framework-bundle.

File: src/Symfony/Component/VarDumper/composer.json (hunk 2 above)
Comment (GromNaN) [reply src/Symfony/Component/VarDumper/composer.json:28]: The code that depends on twig is very robust.

https://github.com/symfony/symfony/blob/d8a4b3c16a08be15aa24e0899bbdcb34c3fef002/src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php#L238

**Review by stof — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**